# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../avahi/dbus-abi.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

dbus ABI compatibility patch

--- avahi-0.6.21/avahi-common/dbus-watch-glue.c.orig	2007-08-12 17:39:21.000000000 +0200
+++ avahi-0.6.21/avahi-common/dbus-watch-glue.c	2007-09-04 08:59:33.000000000 +0200
@@ -134,7 +134,11 @@
 
         if (!(avahi_watch = poll_api->watch_new(
                   poll_api,
+#if (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MINOR == 1 && DBUS_VERSION_MICRO >= 1) || (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MINOR > 1) || (DBUS_VERSION_MAJOR > 1)
                   dbus_watch_get_unix_fd(dbus_watch),
+#else
+                  dbus_watch_get_fd(dbus_watch),
+#endif
                   translate_dbus_to_avahi(dbus_watch_get_flags(dbus_watch)),
                   watch_callback,
                   dbus_watch)))
